packages/page-broker/src/Overview/Workload.tsx

```diff
@@ -3,76 +3,105 @@
 import type { CoreWorkloadType, CoreWorkplanType, InfoRow } from '../types.js';
 
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 
 import { ExpandButton } from '@polkadot/react-components';
-import { useRegions, useToggle } from '@polkadot/react-hooks';
+import { useApi, useBrokerSalesInfo, useCall, useRegions, useToggle } from '@polkadot/react-hooks';
 
 import { formatRowInfo } from '../utils.js';
 import WorkInfoRow from './WorkInfoRow.js';
 import Workplan from './Workplan.js';
 
 interface Props {
   api: ApiPromise;
-  value: CoreWorkloadType
-  timeslice: number;
-  workplan?: CoreWorkplanType[] | null
+  core: number;
+  workload: CoreWorkloadType[] | undefined
+  workplan?: CoreWorkplanType[] | undefined
 }
 
-function Workload ({ api, timeslice, value: { core, info, lastBlock, type }, workplan }: Props): React.ReactElement<Props> {
+function Workload ({ api, core, workload, workplan }: Props): React.ReactElement<Props> {
+  const { isApiReady } = useApi();
+  const salesInfo = useBrokerSalesInfo();
+
+  const status = useCall<Option<PalletBrokerStatusRecord>>(isApiReady && api.query.broker?.status);
   const [isExpanded, toggleIsExpanded] = useToggle(false);
-  const [tableData, setTableData] = useState<InfoRow>();
-  const [currentRegion, setCurrentRegion] = useState<RegionInfo | undefined>();
+  const [workloadData, setWorkloadData] = useState<InfoRow[]>();
+  const [workplanData, setWorkplanData] = useState<InfoRow[]>();
+
+  const currentTimeSlice = useMemo(() => {
+    if (!!status && !!status?.toHuman()) {
+      return status.unwrap().lastCommittedTimeslice.toNumber();
+    }
+
+    return 0;
+  }, [status]);
+
   const regionInfo = useRegions(api);
+  const region: RegionInfo | undefined = useMemo(() => regionInfo?.find((v) => v.core === core && v.start <= currentTimeSlice && v.end > currentTimeSlice), [regionInfo, core, currentTimeSlice]);
 
   useEffect(() => {
-    if (info) {
-      const region: RegionInfo | undefined = regionInfo?.find((v) => v.core === core && v.start <= timeslice && v.end > timeslice);
+    if (!!workload?.length && !!salesInfo) {
+      setWorkloadData(formatRowInfo(workload, core, region, currentTimeSlice, salesInfo));
+    } else {
+      return setWorkloadData([{ core }]);
+    }
+  }, [workload, region, currentTimeSlice, core, salesInfo]);
 
-      setTableData(formatRowInfo(info, core, region, timeslice, type, lastBlock));
-      setCurrentRegion(region);
+  useEffect(() => {
+    if (!!workplan?.length && !!salesInfo) {
+      setWorkplanData(formatRowInfo(workplan, core, region, currentTimeSlice, salesInfo));
     }
-  }, [info, regionInfo, core, timeslice, lastBlock, type]);
+  }
+  , [workplan, region, currentTimeSlice, core, salesInfo]);
 
-  const hasWorkplan = !!workplan?.length;
+  const hasWorkplan = workplan?.length;
 
   return (
     <>
-      {tableData &&
+      {!!workloadData &&
```

Comment:
This can just be `workloadData`. Also if this value has the possibility of being false, it should probably be noted in the types in its `useState`.